### PR TITLE
Adds Dwarven Brewing for the Bartender, to replace the Fermenting Barrel.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Re-adds Modular Brewing from the long dead Lava Dwarfs PR. Replaces the old, lame Fermenting Barrel with this cool and good new Brewing Barrel. It didn't allow you to get anything new anyways, just printed out old existing reagents that you could either mix at the bartender machines, or just get by grinding up the reagents.

Plus, it also allows you to get seeds back from plants. Therefore, people can build ghetto low tech farms without needing to build a seed extractor.

Here's a video of it.
https://file.house/RKM_.mp4

And here's a picture.

## Why It's Good For The Game

It rocks super hard. Make custom brews out of any reagent in the game. Compress reagents for stealthier chemistry payloads. Compress medicine for cooler healing chemicals. All kinds of fun stuff. People really loved this during the test merges of Lava Dwarves.

## Changelog
:cl: Iamgoofball
add: Added the new Brewing Barrel! Make fully custom ales in classic Lava Dwarf fashion! Build with the same recipe as the now removed fermenting barrel.
del: Replaced the old Fermenting Barrel with the new Brewing Barrel.
/:cl:

Closes #48887 
This is made irrelevant due to the Fermenting Barrel being reworked.

### CODE REVIEW

**Comment by trollbreeder** [comment]
The return of the king

**Comment by Iamgoofball** [comment]
what is this new fangled linter stuff and how do i fix it

**Comment by TheChosenEvilOne** [comment]
`file ./code/modules/hydroponics/growninedible.dm is missing a trailing newline`

**Comment by trollbreeder** [comment]
Oh no the goofcode is already falling apart

**Comment by Tharcoonvagh** [comment]
!!BREWING!!

**Comment by Tlaltecuhtli** [comment]
special brews should still be in, like wheat into beer instead of wheat ale

**Comment by Iamgoofball** [comment]
@Tlaltecuhtli the booze dispenser already allows you to get the crappy naddy lite space beer though

**Comment by SpinnerMaster** [comment]
Ghetto hooch? Hell yes.

**Comment by Farquaar** [comment]
Does the fermenting barrel need to be replaced? It's the only low-tech way to produce a number of alcoholic beverages.

**Comment by FantasticFwoosh** [comment]
:point_down: we've been here before Goof, we already know Fel's is the mechanically better, when instead you could have made a subtype, you opted to write all over and remove the fermenting barrel.

Vivadel is already working upon a subtype for Hippiestation's project and you've made absolutely no development or growth upon the idea since you last attempted it since it looks and apparently acts exactly the same. :-1:

**Comment by optimumtact** [comment]
Here' fwoosh, got a video for you
https://www.youtube.com/watch?v=X6jtVon1BJE

until you've watched it and understand, buzz off.

**Comment by AnthonyJNgo** [comment]
This is a great PR, thanks goof. I'm actually still working on a version of dwarves, and this is definitely something extremely useful that I can put to use at least.

**Comment by necromanceranne** [comment]
My peanut brain is already ticking for what possibilities this might present. Will all my kill mixes now involve booze? We will see.

**Comment by Tlaltecuhtli** [comment]
vg's brewer makes you able to choose between wine,ale etc give you a cool bottle with the same color scheme of the seed and transfers parts of the plant reagent into the bottle

**Comment by fludd12** [comment]
I completely approve, I've been considering completely stripping out the "make existing reagents" thing I did for a *while* now, but never got around to it. Custom alcohols are so much cooler for botany and don't tread too hard on the bartender's feet.

File: code/game/atoms.dm
Comment (fludd12) [inline code/game/atoms.dm]: Instead of adding a new proc to the atom level, would a brewable component that defines these behaviors be more desirable?

File: code/modules/food_and_drinks/food.dm
Comment (fludd12) [inline code/modules/food_and_drinks/food.dm]: See above - componentization would cut down on the number of variables added to various locations, and allow you to give non-foods winepower.

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (fludd12) [inline code/modules/hydroponics/fermenting_barrel.dm]: If you think it's a good idea, you could make a separate craft-able barrel that *just* has the reagent storage mechanisms applied to retain the craft-able "large barrel" format. It'd also keep the fancy sprite that was made for me in-game, which I adore.

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (fludd12) [inline code/modules/hydroponics/fermenting_barrel.dm]: Could this be a more self-documenting variable name?

File: code/modules/hydroponics/growninedible.dm
Comment (fludd12) [inline code/modules/hydroponics/growninedible.dm]: IIRC, there is an existing reagents proc to do this automatically.

**Review by fludd12 — COMMENTED** [review]
So much better than the cheap knockoff version I made, just a few things to consider.

**Comment by boodaliboo** [comment]
WELCOME BACK

**Comment by Iamgoofball** [comment]
@FantasticFwoosh it looks and acts exactly the same because this is the version people liked

File: code/modules/food_and_drinks/food.dm
Comment (Iamgoofball) [reply code/modules/food_and_drinks/food.dm]: Since GetComponent isn't allowed I don't know how to feasibly do this.

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (Iamgoofball) [reply code/modules/hydroponics/fermenting_barrel.dm]: its a temp holder var so idk what to name it

File: code/modules/hydroponics/growninedible.dm
Comment (Iamgoofball) [reply code/modules/hydroponics/growninedible.dm]: i dont know the name of said proc

File: code/game/atoms.dm
Comment (Iamgoofball) [reply code/game/atoms.dm]: see below

File: code/modules/food_and_drinks/food.dm
Comment (ninjanomnom) [reply code/modules/food_and_drinks/food.dm]: A component isn't the answer here but a signal can do this nicely in a generic way.

Signals are best applied when there's a single source and it describes the event happening rather than what should be the result of the signal call. You can do both here by replacing the call to the on_brew proc with a signal something like this

```
var/list/brew_results = list()
var/brew_flags = SEND_SIGNAL(thing, COMSIG_ATOM_ON_BREW, brew_results)
...dostuff
if(flags & COMPONENT_BREW_CONSUME)
  qdel(thing)
```

Lists passed in with signals and edited by the receiver also get edited outside the signal so you can use the changes elsewhere. This in a generic way allows for things to specify what results they want from brewing. Note that I had the signal also return a flag to qdel so that by default nothing happens. You'll probably also want nothing to happen if no flags and no brew results are given back out.

**Comment by MCterra10** [comment]
Lol @ fwoosh downredditing all the comments, what a salty little fellow

File: code/__DEFINES/components.dm
Comment (fludd12) [inline code/__DEFINES/components.dm]: ~~sorry for what~~
why do you do this. just do what everything else does.

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (fludd12) [reply code/modules/hydroponics/fermenting_barrel.dm]: Why not, like, "addreagents" or something?

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm
Comment (fludd12) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: I will say, I am questioning whether or not the contained reagents system is a good idea to keep moving forward with. I'll leave it to actual maintainers to decide, but keep in mind that this work may be scrapped. I do like the rest of the custom alcohol system, though.

**Review by fludd12 — COMMENTED** [review]
More 🚲🏚 from I, the master bikeshedder

File: code/__DEFINES/components.dm
Comment (Arturlang) [reply code/__DEFINES/components.dm]: linter failed due to no newline here, just add it and travis should be content

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm
Comment (Iamgoofball) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: This will not be scrapped, as it is the entire point of the gimmick.

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (Iamgoofball) [reply code/modules/hydroponics/fermenting_barrel.dm]: done

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (Iamgoofball) [reply code/modules/hydroponics/fermenting_barrel.dm]: i don't really see the utility in this when one is available via plumbing already

the barrel sprite is nice tho, i'll see about finding a use for it

File: code/modules/food_and_drinks/food.dm
Comment (Iamgoofball) [reply code/modules/food_and_drinks/food.dm]: resolved, we're using 2020 signals now, goofcode in the future :tm:

**Comment by carshalash** [comment]
Every drink from it is ale huh?

**Comment by ArcaneMusic** [comment]
In any case, you should also throw the brewing barrel (or the storage barrel if you go with that) into the stack crafting menu for wood, since they still just use wood planks at the moment, and can only be made under the tribal crafting menu, which is a fairly obscure place to look for it.

**Comment by Iamgoofball** [comment]
Stack crafting should be removed and migrated to the crafting menu.

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (ninjanomnom) [inline code/modules/hydroponics/fermenting_barrel.dm]: Why are you returning false for regular reagent containers?

File: code/modules/hydroponics/fermenting_barrel.dm
Comment (ninjanomnom) [inline code/modules/hydroponics/fermenting_barrel.dm]: pass in src so you can do the other reviews

File: code/modules/food_and_drinks/food.dm
Comment (ninjanomnom) [inline code/modules/food_and_drinks/food.dm]: If you want to return multiple values like this you're better off modifying the barrel from the signal receivers. This sort of thing makes for error prone code. Make some vars on the barrel to keep track of booze power and prefix then just modify them here

File: code/modules/hydroponics/seed_extractor.dm
Comment (ninjanomnom) [inline code/modules/hydroponics/seed_extractor.dm]:
```suggestion
	var/seedloc = O.drop_location()

```

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm
Comment (ninjanomnom) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]:
```suggestion
	. = ..()
```

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm
Comment (ninjanomnom) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]:
```suggestion
	return ..()
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/food.dm
Comment (Iamgoofball) [reply code/modules/food_and_drinks/food.dm]: what? No, go read how the barrel works

**Comment by ExcessiveUseOfCobblestone** [comment]
goof needs to unblock me so I can review the PR or i'm going to close it 8)

File: code/modules/food_and_drinks/food.dm
Comment (Qustinnus) [reply code/modules/food_and_drinks/food.dm]: @ninjanomnom adding all this brewery shit is gonna make the edible component incompatible. Shouldn't this brew shit be in a component?

File: code/modules/food_and_drinks/food.dm
Comment (Qustinnus) [reply code/modules/food_and_drinks/food.dm]: ditto, this should be in a component and not on food itself, seeing as we're removing /food soon

Comment (fludd12) [reply code/modules/food_and_drinks/food.dm]: See ninja's explanation for why components aren't optimal earlier, though I'm with you on the idea that it'd be nice to have some external handler for these functions instead of just signal registering.

Comment (Qustinnus) [reply code/modules/food_and_drinks/food.dm]: then elements should be used. It's stupid to have this shit on the /food subtype when we're trying to kill it and it just means I have to clean it up in the future.

**Comment by Iamgoofball** [comment]
Unblocked, but if you violate the Github ToS on my pull requests going forward I will not hesitate to report you.

**Comment by Qustinnus** [comment]
posted by: man who breaks ToS every odd PR

**Comment by optimumtact** [comment]
I see no reason to allow you to contribute if you are going to actively threaten my maintainers with reporting them to github for any perceived infraction.

I don't want to deal with you if you're going to act like a landmine.

Repeat that behaviour again and you'll be the one blocked from this repo.

**Comment by Iamgoofball** [comment]
I already spoke with ninja about this in coderbus because I had Cobby blocked from well before they were a maintainer, as they were posting harassing comments on my pull requests, and the headcoders/maintainers at the time had advised me to block users instead of requesting their comments be deleted.

Apologies for coming off rude in that comment but I perceived this as a maintainer using their position of authority to bypass a block that I had placed to protect myself from harassment. It is clear now that this was not the case, and all involved parties should of phrased things differently.

**Comment by Ty-the-Smonk** [comment]
Sad, this looked promising to.

**Comment by Iamgoofball** [comment]
Take it up with oranges, feature is denied until he rules otherwise.

**Pull request closed without merging**
